diff --git a/node_modules/y-quill/src/y-quill.js b/node_modules/y-quill/src/y-quill.js
index e668115..684f889 100644
--- a/node_modules/y-quill/src/y-quill.js
+++ b/node_modules/y-quill/src/y-quill.js
@@ -122,11 +122,7 @@ export class QuillBinding {
       if (awareness && quillCursors) {
         const sel = quill.getSelection()
         const aw = /** @type {any} */ (awareness.getLocalState())
-        if (sel === null) {
-          if (awareness.getLocalState() !== null) {
-            awareness.setLocalStateField('cursor', /** @type {any} */ (null))
-          }
-        } else {
+        if (sel !== null) {
           const anchor = Y.createRelativePositionFromTypeIndex(type, sel.index)
           const head = Y.createRelativePositionFromTypeIndex(type, sel.index + sel.length)
           if (!aw || !aw.cursor || !Y.compareRelativePositions(anchor, aw.cursor.anchor) || !Y.compareRelativePositions(head, aw.cursor.head)) {
@@ -155,6 +151,16 @@ export class QuillBinding {
       })
       awareness.on('change', this._awarenessChange)
     }
+
+    let focused = false
+    quill.on('selection-change', (range) => {
+      if (range) {
+        focused = true
+      } else if (focused) {
+        focused = false
+        awareness.setLocalStateField('cursor', null)
+      }
+    })
   }
   destroy () {
     this.type.unobserve(this._typeObserver)
