diff --git a/node_modules/y-websocket/bin/utils.js b/node_modules/y-websocket/bin/utils.js
index 0c9a7b7..15985ca 100644
--- a/node_modules/y-websocket/bin/utils.js
+++ b/node_modules/y-websocket/bin/utils.js
@@ -36,9 +36,9 @@ if (typeof persistenceDir === 'string') {
     provider: ldb,
     bindState: async (docName, ydoc) => {
       const persistedYdoc = await ldb.getYDoc(docName)
-      const newUpdates = Y.encodeStateAsUpdate(ydoc)
+      const newUpdates = Y.encodeStateAsUpdateV2(ydoc)
       ldb.storeUpdate(docName, newUpdates)
-      Y.applyUpdate(ydoc, Y.encodeStateAsUpdate(persistedYdoc))
+      Y.applyUpdateV2(ydoc, Y.encodeStateAsUpdateV2(persistedYdoc))
       ydoc.on('update', update => {
         ldb.storeUpdate(docName, update)
       })
